pkg/cmd/cli/cmd/env.go

```diff
@@ -349,36 +361,38 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 	return nil
 }
 
-func updateEnv(existing []kapi.EnvVar, env []kapi.EnvVar, remove []string) []kapi.EnvVar {
+func updateEnv(existing []kapi.EnvVar, env []kapi.EnvVar, remove []string) ([]kapi.EnvVar, bool) {
 	out := []kapi.EnvVar{}
-	covered := sets.NewString(remove...)
+	envUpdated := false
+	toDelete := sets.NewString(remove...)
+
 	for _, e := range existing {
-		if covered.Has(e.Name) {
+		if toDelete.Has(e.Name) {
+			envUpdated = true
 			continue
 		}
-		newer, ok := findEnv(env, e.Name)
+		newer, ok := findEnv(env, e)
 		if ok {
-			covered.Insert(e.Name)
-			out = append(out, newer)
-			continue
+			envUpdated = true
 		}
-		out = append(out, e)
+		out = append(out, newer)
 	}
+
 	for _, e := range env {
-		if covered.Has(e.Name) {
+		if toDelete.Has(e.Name) {
 			continue
 		}
-		covered.Insert(e.Name)
 		out = append(out, e)
+		envUpdated = true
 	}
-	return out
+	return out, envUpdated
 }
 
-func findEnv(env []kapi.EnvVar, name string) (kapi.EnvVar, bool) {
-	for _, e := range env {
-		if e.Name == name {
+func findEnv(envs []kapi.EnvVar, env kapi.EnvVar) (kapi.EnvVar, bool) {
+	for _, e := range envs {
+		if e.Name == env.Name {
 			return e, true
 		}
 	}
-	return kapi.EnvVar{}, false
+	return env, false
```

Comment:
the old way was clearer... putting a value in and not easily knowing which you get back is confusing